File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/oac_object_column.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class OacObjectColumn(object):
    """
    Column metadata for a table exposed by an Oracle Analytics object.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new OacObjectColumn object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param name:
            The value to assign to the name property of this OacObjectColumn.
        :type name: str

        :param description:
            The value to assign to the description property of this OacObjectColumn.
        :type description: str

        :param formula:
            The value to assign to the formula property of this OacObjectColumn.
        :type formula: str

        :param is_time:
            The value to assign to the is_time property of this OacObjectColumn.
        :type is_time: bool

        :param type:
            The value to assign to the type property of this OacObjectColumn.
        :type type: str

        :param hierarchy_id:
            The value to assign to the hierarchy_id property of this OacObjectColumn.
        :type hierarchy_id: str

        :param hierarchy_display_name:
            The value to assign to the hierarchy_display_name property of this OacObjectColumn.
        :type hierarchy_display_name: str

        :param dimension_id:
            The value to assign to the dimension_id property of this OacObjectColumn.
        :type dimension_id: str

        :param table_name:
            The value to assign to the table_name property of this OacObjectColumn.
        :type table_name: str

        :param levels:
            The value to assign to the levels property of this OacObjectColumn.
        :type levels: list[oci.aidataplatform_dp.models.OacObjectHierarchyLevel]

        :param sql_info:
            The value to assign to the sql_info property of this OacObjectColumn.
        :type sql_info: oci.aidataplatform_dp.models.OacObjectColumnSqlInfo

        """
        self.swagger_types = {
            'name': 'str',
            'description': 'str',
            'formula': 'str',
            'is_time': 'bool',
            'type': 'str',
            'hierarchy_id': 'str',
            'hierarchy_display_name': 'str',
            'dimension_id': 'str',
            'table_name': 'str',
            'levels': 'list[OacObjectHierarchyLevel]',
            'sql_info': 'OacObjectColumnSqlInfo'
        }

        self.attribute_map = {
            'name': 'name',
            'description': 'description',
            'formula': 'formula',
            'is_time': 'isTime',
            'type': 'type',
            'hierarchy_id': 'hierarchyID',
            'hierarchy_display_name': 'hierarchyDisplayName',
            'dimension_id': 'dimensionID',
            'table_name': 'tableName',
            'levels': 'levels',
            'sql_info': 'sqlInfo'
        }

        self._name = None
        self._description = None
        self._formula = None
        self._is_time = None
        self._type = None
        self._hierarchy_id = None
        self._hierarchy_display_name = None
        self._dimension_id = None
        self._table_name = None
        self._levels = None
        self._sql_info = None

    @property
    def name(self):
        """
        Gets the name of this OacObjectColumn.
        The Oracle Analytics column name.


        :return: The name of this OacObjectColumn.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this OacObjectColumn.
        The Oracle Analytics column name.


        :param name: The name of this OacObjectColumn.
        :type: str
        """
        self._name = name

    @property
    def description(self):
        """
        Gets the description of this OacObjectColumn.
        The Oracle Analytics column description.


        :return: The description of this OacObjectColumn.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this OacObjectColumn.
        The Oracle Analytics column description.


        :param description: The description of this OacObjectColumn.
        :type: str
        """
        self._description = description

    @property
    def formula(self):
        """
        Gets the formula of this OacObjectColumn.
        The Oracle Analytics expression backing the column.


        :return: The formula of this OacObjectColumn.
        :rtype: str
        """
        return self._formula

    @formula.setter
    def formula(self, formula):
        """
        Sets the formula of this OacObjectColumn.
        The Oracle Analytics expression backing the column.


        :param formula: The formula of this OacObjectColumn.
        :type: str
        """
        self._formula = formula

    @property
    def is_time(self):
        """
        Gets the is_time of this OacObjectColumn.
        Indicates whether this Oracle Analytics column represents time.


        :return: The is_time of this OacObjectColumn.
        :rtype: bool
        """
        return self._is_time

    @is_time.setter
    def is_time(self, is_time):
        """
        Sets the is_time of this OacObjectColumn.
        Indicates whether this Oracle Analytics column represents time.


        :param is_time: The is_time of this OacObjectColumn.
        :type: bool
        """
        self._is_time = is_time

    @property
    def type(self):
        """
        Gets the type of this OacObjectColumn.
        The Oracle Analytics column type.


        :return: The type of this OacObjectColumn.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this OacObjectColumn.
        The Oracle Analytics column type.


        :param type: The type of this OacObjectColumn.
        :type: str
        """
        self._type = type

    @property
    def hierarchy_id(self):
        """
        Gets the hierarchy_id of this OacObjectColumn.
        The hierarchy identifier reported for an Oracle Analytics hierarchical column.


        :return: The hierarchy_id of this OacObjectColumn.
        :rtype: str
        """
        return self._hierarchy_id

    @hierarchy_id.setter
    def hierarchy_id(self, hierarchy_id):
        """
        Sets the hierarchy_id of this OacObjectColumn.
        The hierarchy identifier reported for an Oracle Analytics hierarchical column.


        :param hierarchy_id: The hierarchy_id of this OacObjectColumn.
        :type: str
        """
        self._hierarchy_id = hierarchy_id

    @property
    def hierarchy_display_name(self):
        """
        Gets the hierarchy_display_name of this OacObjectColumn.
        The hierarchy display name reported for a hierarchical column in an OAC data object.


        :return: The hierarchy_display_name of this OacObjectColumn.
        :rtype: str
        """
        return self._hierarchy_display_name

    @hierarchy_display_name.setter
    def hierarchy_display_name(self, hierarchy_display_name):
        """
        Sets the hierarchy_display_name of this OacObjectColumn.
        The hierarchy display name reported for a hierarchical column in an OAC data object.


        :param hierarchy_display_name: The hierarchy_display_name of this OacObjectColumn.
        :type: str
        """
        self._hierarchy_display_name = hierarchy_display_name

    @property
    def dimension_id(self):
        """
        Gets the dimension_id of this OacObjectColumn.
        The dimension identifier reported for an Oracle Analytics hierarchical column.


        :return: The dimension_id of this OacObjectColumn.
        :rtype: str
        """
        return self._dimension_id

    @dimension_id.setter
    def dimension_id(self, dimension_id):
        """
        Sets the dimension_id of this OacObjectColumn.
        The dimension identifier reported for an Oracle Analytics hierarchical column.


        :param dimension_id: The dimension_id of this OacObjectColumn.
        :type: str
        """
        self._dimension_id = dimension_id

    @property
    def table_name(self):
        """
        Gets the table_name of this OacObjectColumn.
        The table name reported for a hierarchical column in an OAC data object.


        :return: The table_name of this OacObjectColumn.
        :rtype: str
        """
        return self._table_name

    @table_name.setter
    def table_name(self, table_name):
        """
        Sets the table_name of this OacObjectColumn.
        The table name reported for a hierarchical column in an OAC data object.


        :param table_name: The table_name of this OacObjectColumn.
        :type: str
        """
        self._table_name = table_name

    @property
    def levels(self):
        """
        Gets the levels of this OacObjectColumn.
        The hierarchy levels reported for an Oracle Analytics hierarchical column.


        :return: The levels of this OacObjectColumn.
        :rtype: list[oci.aidataplatform_dp.models.OacObjectHierarchyLevel]
        """
        return self._levels

    @levels.setter
    def levels(self, levels):
        """
        Sets the levels of this OacObjectColumn.
        The hierarchy levels reported for an Oracle Analytics hierarchical column.


        :param levels: The levels of this OacObjectColumn.
        :type: list[oci.aidataplatform_dp.models.OacObjectHierarchyLevel]
        """
        self._levels = levels

    @property
    def sql_info(self):
        """
        Gets the sql_info of this OacObjectColumn.

        :return: The sql_info of this OacObjectColumn.
        :rtype: oci.aidataplatform_dp.models.OacObjectColumnSqlInfo
        """
        return self._sql_info

    @sql_info.setter
    def sql_info(self, sql_info):
        """
        Sets the sql_info of this OacObjectColumn.

        :param sql_info: The sql_info of this OacObjectColumn.
        :type: oci.aidataplatform_dp.models.OacObjectColumnSqlInfo
        """
        self._sql_info = sql_info

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/OacObjectColumn.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Column metadata for a table exposed by an Oracle Analytics object.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=OacObjectColumn.Builder.class)

public final class OacObjectColumn  {
    @Deprecated
    @java.beans.ConstructorProperties({"name", "description", "formula", "isTime", "type", "hierarchyID", "hierarchyDisplayName", "dimensionID", "tableName", "levels", "sqlInfo"})
    public OacObjectColumn(String name, String description, String formula, Boolean isTime, String type, String hierarchyID, String hierarchyDisplayName, String dimensionID, String tableName, java.util.List<OacObjectHierarchyLevel> levels, OacObjectColumnSqlInfo sqlInfo) {
        super();
        this.name = name;
        this.description = description;
        this.formula = formula;
        this.isTime = isTime;
        this.type = type;
        this.hierarchyID = hierarchyID;
        this.hierarchyDisplayName = hierarchyDisplayName;
        this.dimensionID = dimensionID;
        this.tableName = tableName;
        this.levels = levels;
        this.sqlInfo = sqlInfo;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The Oracle Analytics column name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;

        /**
         * The Oracle Analytics column name.
         * @param name the value to set
         * @return this builder
         **/
        

public Builder name(String name) {
    this.name = name;
    return this;
}
            /**
     * The Oracle Analytics column description.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * The Oracle Analytics column description.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
            /**
     * The Oracle Analytics expression backing the column.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("formula")
private String formula;

        /**
         * The Oracle Analytics expression backing the column.
         * @param formula the value to set
         * @return this builder
         **/
        

public Builder formula(String formula) {
    this.formula = formula;
    return this;
}
            /**
     * Indicates whether this Oracle Analytics column represents time.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isTime")
private Boolean isTime;

        /**
         * Indicates whether this Oracle Analytics column represents time.
         * @param isTime the value to set
         * @return this builder
         **/
        

public Builder isTime(Boolean isTime) {
    this.isTime = isTime;
    return this;
}
            /**
     * The Oracle Analytics column type.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private String type;

        /**
         * The Oracle Analytics column type.
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(String type) {
    this.type = type;
    return this;
}
            /**
     * The hierarchy identifier reported for an Oracle Analytics hierarchical column.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("hierarchyID")
private String hierarchyID;

        /**
         * The hierarchy identifier reported for an Oracle Analytics hierarchical column.
         * @param hierarchyID the value to set
         * @return this builder
         **/
        

public Builder hierarchyID(String hierarchyID) {
    this.hierarchyID = hierarchyID;
    return this;
}
            /**
     * The hierarchy display name reported for a hierarchical column in an OAC data object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("hierarchyDisplayName")
private String hierarchyDisplayName;

        /**
         * The hierarchy display name reported for a hierarchical column in an OAC data object.
         * @param hierarchyDisplayName the value to set
         * @return this builder
         **/
        

public Builder hierarchyDisplayName(String hierarchyDisplayName) {
    this.hierarchyDisplayName = hierarchyDisplayName;
    return this;
}
            /**
     * The dimension identifier reported for an Oracle Analytics hierarchical column.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("dimensionID")
private String dimensionID;

        /**
         * The dimension identifier reported for an Oracle Analytics hierarchical column.
         * @param dimensionID the value to set
         * @return this builder
         **/
        

public Builder dimensionID(String dimensionID) {
    this.dimensionID = dimensionID;
    return this;
}
            /**
     * The table name reported for a hierarchical column in an OAC data object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("tableName")
private String tableName;

        /**
         * The table name reported for a hierarchical column in an OAC data object.
         * @param tableName the value to set
         * @return this builder
         **/
        

public Builder tableName(String tableName) {
    this.tableName = tableName;
    return this;
}
            /**
     * The hierarchy levels reported for an Oracle Analytics hierarchical column.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("levels")
private java.util.List<OacObjectHierarchyLevel> levels;

        /**
         * The hierarchy levels reported for an Oracle Analytics hierarchical column.
         * @param levels the value to set
         * @return this builder
         **/
        

public Builder levels(java.util.List<OacObjectHierarchyLevel> levels) {
    this.levels = levels;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("sqlInfo")
private OacObjectColumnSqlInfo sqlInfo;



public Builder sqlInfo(OacObjectColumnSqlInfo sqlInfo) {
    this.sqlInfo = sqlInfo;
    return this;
}


        public OacObjectColumn build() {
            OacObjectColumn model = new OacObjectColumn(this.name
                , this.description
                , this.formula
                , this.isTime
                , this.type
                , this.hierarchyID
                , this.hierarchyDisplayName
                , this.dimensionID
                , this.tableName
                , this.levels
                , this.sqlInfo);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(OacObjectColumn model) {
                this.name(model.getName());
    this.description(model.getDescription());
    this.formula(model.getFormula());
    this.isTime(model.getIsTime());
    this.type(model.getType());
    this.hierarchyID(model.getHierarchyID());
    this.hierarchyDisplayName(model.getHierarchyDisplayName());
    this.dimensionID(model.getDimensionID());
    this.tableName(model.getTableName());
    this.levels(model.getLevels());
    this.sqlInfo(model.getSqlInfo());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The Oracle Analytics column name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * The Oracle Analytics column name.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * The Oracle Analytics column description.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * The Oracle Analytics column description.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * The Oracle Analytics expression backing the column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("formula")
    private final String formula;

        /**
     * The Oracle Analytics expression backing the column.
     * @return the value
     **/
    
    public String getFormula() {
        return formula;
    }


        /**
     * Indicates whether this Oracle Analytics column represents time.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isTime")
    private final Boolean isTime;

        /**
     * Indicates whether this Oracle Analytics column represents time.
     * @return the value
     **/
    
    public Boolean getIsTime() {
        return isTime;
    }


        /**
     * The Oracle Analytics column type.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final String type;

        /**
     * The Oracle Analytics column type.
     * @return the value
     **/
    
    public String getType() {
        return type;
    }


        /**
     * The hierarchy identifier reported for an Oracle Analytics hierarchical column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("hierarchyID")
    private final String hierarchyID;

        /**
     * The hierarchy identifier reported for an Oracle Analytics hierarchical column.
     * @return the value
     **/
    
    public String getHierarchyID() {
        return hierarchyID;
    }


        /**
     * The hierarchy display name reported for a hierarchical column in an OAC data object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("hierarchyDisplayName")
    private final String hierarchyDisplayName;

        /**
     * The hierarchy display name reported for a hierarchical column in an OAC data object.
     * @return the value
     **/
    
    public String getHierarchyDisplayName() {
        return hierarchyDisplayName;
    }


        /**
     * The dimension identifier reported for an Oracle Analytics hierarchical column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("dimensionID")
    private final String dimensionID;

        /**
     * The dimension identifier reported for an Oracle Analytics hierarchical column.
     * @return the value
     **/
    
    public String getDimensionID() {
        return dimensionID;
    }


        /**
     * The table name reported for a hierarchical column in an OAC data object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tableName")
    private final String tableName;

        /**
     * The table name reported for a hierarchical column in an OAC data object.
     * @return the value
     **/
    
    public String getTableName() {
        return tableName;
    }


        /**
     * The hierarchy levels reported for an Oracle Analytics hierarchical column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("levels")
    private final java.util.List<OacObjectHierarchyLevel> levels;

        /**
     * The hierarchy levels reported for an Oracle Analytics hierarchical column.
     * @return the value
     **/
    
    public java.util.List<OacObjectHierarchyLevel> getLevels() {
        return levels;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("sqlInfo")
    private final OacObjectColumnSqlInfo sqlInfo;

    
    public OacObjectColumnSqlInfo getSqlInfo() {
        return sqlInfo;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("OacObjectColumn(");
        sb.append("name=").append(String.valueOf(this.name));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", formula=").append(String.valueOf(this.formula));
        sb.append(", isTime=").append(String.valueOf(this.isTime));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", hierarchyID=").append(String.valueOf(this.hierarchyID));
        sb.append(", hierarchyDisplayName=").append(String.valueOf(this.hierarchyDisplayName));
        sb.append(", dimensionID=").append(String.valueOf(this.dimensionID));
        sb.append(", tableName=").append(String.valueOf(this.tableName));
        sb.append(", levels=").append(String.valueOf(this.levels));
        sb.append(", sqlInfo=").append(String.valueOf(this.sqlInfo));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof OacObjectColumn)) {
            return false;
        }

        OacObjectColumn other = (OacObjectColumn) o;
        return java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.formula, other.formula) &&
            java.util.Objects.equals(this.isTime, other.isTime) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.hierarchyID, other.hierarchyID) &&
            java.util.Objects.equals(this.hierarchyDisplayName, other.hierarchyDisplayName) &&
            java.util.Objects.equals(this.dimensionID, other.dimensionID) &&
            java.util.Objects.equals(this.tableName, other.tableName) &&
            java.util.Objects.equals(this.levels, other.levels) &&
            java.util.Objects.equals(this.sqlInfo, other.sqlInfo);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.formula == null ? 43 : this.formula.hashCode());
        result = (result * PRIME) + (this.isTime == null ? 43 : this.isTime.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.hierarchyID == null ? 43 : this.hierarchyID.hashCode());
        result = (result * PRIME) + (this.hierarchyDisplayName == null ? 43 : this.hierarchyDisplayName.hashCode());
        result = (result * PRIME) + (this.dimensionID == null ? 43 : this.dimensionID.hashCode());
        result = (result * PRIME) + (this.tableName == null ? 43 : this.tableName.hashCode());
        result = (result * PRIME) + (this.levels == null ? 43 : this.levels.hashCode());
        result = (result * PRIME) + (this.sqlInfo == null ? 43 : this.sqlInfo.hashCode());
        return result;
    }


}
